File: poo/Cpp/ManipulandoArrays/ManipulandoArrays.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
    int n, m;
    std::cin >> n >> m;

    std::vector<long long> diff(n + 1, 0);

    for (int i = 0; i < m; ++i) {
        int a, b;
        long long k;
        std::cin >> a >> b >> k;
        diff[a - 1] += k;
        if (b < n) {
            diff[b] -= k;
        }
    }

    long long max_val = 0;
    long long current_val = 0;
    for (int i = 0; i < n; ++i) {
        current_val += diff[i];
        if (current_val > max_val) {
            max_val = current_val;
        }
    }

    std::cout << max_val << std::endl;
    std::cout << std::endl;

    return 0;
}

File: poo/Java/ManipulandoArrays/ManipulandoArrays.java
import java.util.Scanner;

public class ManipulandoArrays {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        long[] diff = new long[n + 1];
        for (int i = 0; i < m; i++) {
            int a = scanner.nextInt();
            int b = scanner.nextInt();
            long k = scanner.nextLong();
            diff[a - 1] += k;
            if (b < n) {
                diff[b] -= k;
            }
        }
        long maxVal = 0;
        long currentVal = 0;
        for (int i = 0; i < n; i++) {
            currentVal += diff[i];
            if (currentVal > maxVal) {
                maxVal = currentVal;
            }
        }
        System.out.println(maxVal);
        System.out.println();
        scanner.close();
    }
}